### PR TITLE
cleanup: SpruceTextAreaWidget

### PR DESCRIPTION
**Pull request by FlashyReese**
I'm tempted to replace your entire switch case with a HashMap.

### CODE REVIEW

File: src/main/java/me/lambdaurora/spruceui/SpruceTextAreaWidget.java [hunk 1]
```diff
@@ -990,4 +971,4 @@ private boolean isInverted()
             return this.anchor.row > this.follower.row || (this.anchor.row == this.follower.row && this.anchor.column > this.follower.column);
         }
     }
-}
+}
```
Comment (LambdAurora) [inline src/main/java/me/lambdaurora/spruceui/SpruceTextAreaWidget.java]: Where's my newline? :<

File: src/main/java/me/lambdaurora/spruceui/SpruceTextAreaWidget.java [hunk 2]
```diff
@@ -662,6 +642,14 @@ public Cursor(boolean main)
             this.main = main;
         }
 
+        public void resetColumn(){
```
Comment (LambdAurora) [inline src/main/java/me/lambdaurora/spruceui/SpruceTextAreaWidget.java]:
```suggestion
        public void toLineStart() {
```

File: src/main/java/me/lambdaurora/spruceui/SpruceTextAreaWidget.java [hunk 3]
```diff
@@ -691,7 +679,7 @@ public void moveHorizontal(int amount)
                 this.column = lines.get(this.row).length();
             }
             if (this.column < 0) {
-                if (this.row == 0) this.column = 0;
+                if (this.row == 0) this.resetColumn();
```
Comment (LambdAurora) [inline src/main/java/me/lambdaurora/spruceui/SpruceTextAreaWidget.java]:
```suggestion
                if (this.row == 0) this.toLineStart();
```

File: src/main/java/me/lambdaurora/spruceui/SpruceTextAreaWidget.java [hunk 4]
```diff
@@ -722,8 +710,8 @@ public void moveVertical(int amount)
 
         public void toStart()
         {
-            this.row = 0;
-            this.column = 0;
+            this.resetRow();
+            this.resetColumn();
```
Comment (LambdAurora) [inline src/main/java/me/lambdaurora/spruceui/SpruceTextAreaWidget.java]:
```suggestion
            this.toLineStart();
```

File: src/main/java/me/lambdaurora/spruceui/SpruceTextAreaWidget.java (hunk 1 above)
Comment (FlashyReese) [reply src/main/java/me/lambdaurora/spruceui/SpruceTextAreaWidget.java]: I swear, I will put it back .-.

**Review by LambdAurora — COMMENTED** [review]
Thanks a lot, maybe rename the `resetColumn` to `toLineStart` sounds a bit better.

Though, for the `HashMap` it sounds a bit horrible, I don't really see how you want to do it.
If you do, please use a fastutil map, hearing Jelly talking about the `HashMap` implementation is still giving  me nightmares.

**Comment by FlashyReese** [comment]
> Though, for the `HashMap` it sounds a bit horrible, I don't really see how you want to do it.
> If you do, please use a fastutil map, hearing Jelly talking about the `HashMap` implementation is still giving me nightmares.

Understandable.

**Comment by FlashyReese** [comment]
I have decided against using a map, it's just going to end up with more unnecessary lines of code.

**Pull request merged**